security/cmd/istio_ca/main.go

```diff
@@ -178,8 +178,8 @@ func init() {
 		defaultWorkloadMinCertGracePeriod, "The minimum workload certificate rotation grace period.")
 
 	// gRPC server for signing CSRs.
-	flags.StringVar(&opts.grpcHostname, "grpc-hostname", "localhost", "The hostname for GRPC server.")
-	flags.IntVar(&opts.grpcPort, "grpc-port", 0, "The port number for GRPC server. "+
+	flags.StringVar(&opts.grpcHostname, "grpc-hostname", "istio-ca", "The hostname for GRPC server.")
```

Comment:
This can be a bit problematic for multi-cloud and mesh expansion, it works today with the dnsmasq hack but will be trickier long term.

Can we use istio-ca.istio-system as default ? Did we implement 'multiple domains' ? If yes, can it be a list.

Reply:
Good point, we have another dir to handle multi-cluster CA. https://github.com/istio/istio/blob/master/security/cmd/multicluster_ca/main.go

Reply:
The issue is irrelevant to this change. 

Currently it does not take a list, and I would like to see what's the requirement before we actually implementing something.

Reply:
@costinm could you clarify what multi-cloud and mesh expansion means? My understanding is 
- Istio service mesh spanning across multi cloud (onprem, gcp)
- one dns name does not make sense in another cloud environmet. e.g istio-ca only regsitered in gke dns server.
- No matter Istio cluster span multi-cloud env, Istio-CA still runs within a single env (AFAIK?)

If so, there should be no issue for this PR. The flag configures the liveness probe controller running within the istio-ca process, so "istio-ca" should just work, even for multi-cloud.

Reply:
Inside a K8S cluster - right now we use secrets, so the name on this endpoint does not matter.

Raw VMs have a node-agent that connect to istio-ca - and rely on a hack, where istio-ca name is mapped to an
internal load balancer address. So making a call to https://istio-ca will work as expected.

However we want to stop requiring users from installing dnsmasq on each VM - and use a DNS server.

Besides, even in K8S  'istiio-ca' name only works for pods in istio-system namespace - no other namespace will be able to reach the node agent. 

Is there any reason to use a unqualified domain name in a DNS record where pretty much everyone uses FQDN ?